src/nimib/types.nim

```diff
@@ -33,6 +33,7 @@ type
     renderPlans*: Table[string, seq[string]]
     renderProcs*: Table[string, NbRenderProc]
     id: int
+    nbJsGlobalScript*, nbJsScript*: string
```

Comment:
We could do it with one script and interlace the block and global scripts like this:
```nim
global1
block:
  block1
block:
  block2
global2
global3
block:
  block3
```
But it would be a different behavior than putting all globals at the top:
```nim
global1
global2
global3
block:
  1
block:
  2
block:
  3
```
as the order of execution changes. So it depends on what we want the global code to be able to do. I like the global-at-the-top approach because it allows us to do customizations of global variables defined by other blocks. Let's say that we have a block which exposes a `config: Table`. Then to configure this block, you modify it inside a `nbJsFromCodeGlobal` block like this:
```nim
nbJsFromCodeGlobal:
  config["param1"] = 3.14
```
The place in the code where this variable change takes place is crucial for if it will take effect or not. If `block1` in the first example is where the hypothetical block reads `config` and starts whatever process it does. And if `global2` is where we change this parameter, then it will be changed *after* it was read and might not make a change. If on the other hand we had all the global at the top like in the second example, `config` would be changed in time before `block1` is run.

This is a very hypothetical example, but it shows a potential use-case which wouldn't be possible if we didn't separate the global and block codes.